python/lsst/ts/MTAirCompressor/aircompressor_csc.py

```diff
@@ -178,51 +169,55 @@ def run_sim():
                 None, run_sim
             )
             await asyncio.sleep(2)
-            self.client = ModbusClient(host="localhost", port=5020)
-        else:
-            self.client = ModbusClient(host=self.hostname)
-
-        if self.client.connect() is False:
-            self.log.error(f"Cannot establish connection to {self.hostname}")
 
-        self.telemetry_task = asyncio.create_task(self.telemetry_loop())
+        try:
+            await self.connect()
+            if self.poll_task.done():
+                self.poll_task = asyncio.create_task(self.poll_loop())
+        except ModbusError as er:
+            await self.log_modbus_error(er, "Starting up:", True)
+            return
 
     async def end_enable(self, data):
         """Power on compressor after switching to enable state. Raise exception
-        if compressor cannot be powered on."""
-        poweredOn = self.client.write_register(
-            Registers.REMOTE_CMD, 0xFF01, unit=self.unit
-        )
-        if poweredOn.isError():
-            raise ModbusError(
-                "Cannot power on compressor", poweredOn, Registers.REMOTE_CMD
+        if compressor cannot be powered on. Ignore state transition triggered
+        by auto update."""
+        if self._status_update is True:
+            return
+        if self._start_by_remote is False:
+            raise RuntimeError(
+                "Compressor isn't in remote mode - cannot be powered on remotely"
```

Comment:
What happens if the compressor is already turned on? Surely you don't want to raise an exception in that case?

Reply:
See above for _status_update. Either it is powered up locally and then exception will not be raised, or it is assumed to be powered on remotely but the flag isn't yet set (by investigation, this can happens https://github.com/lsst-ts/ts_MTAirCompressor/pull/6/commits/fde98e2f7a33218346893277c0b44d6f87a20c91